lib/src/working_copy.rs

```diff
@@ -48,6 +49,7 @@ use crate::tree_builder::TreeBuilder;
 pub enum FileType {
     Normal { executable: bool },
     Symlink,
+    GitSubmodule { id: CommitId },
```

Comment:
Do we need the id? It seems unused as far as I can tell (just being passed around a bit).